packages/react-components/src/icons/src/Icon.tsx

```diff
@@ -60,27 +73,31 @@ export function InnerIcon(props) {
             )}
         />
     );
-}
+}) as InnerIconComponentWithFactory;
 
-InnerIcon.propTypes = propTypes;
+type IconComponentWithFactory = OrbitComponent<"svg", InnerIconProps> & {
+    create: IconFactory
+}
```

Comment:
I would replace all this `Icon.create` thing by a `createIcon` function, what do you think?

Would also requires to update the docs pages "?path=/docs/icon--default-story"
- update the usage
- update the imports to add `createIcon`

Reply:
Oh we would also need a `createMultivariantIcon`